modules/shodan_tools.py

```diff
@@ -1,32 +1,58 @@
 import os
 import json
+import socket
 
 from shodan import Shodan
 from config import SHODAN_API_KEY
 
 api = Shodan(SHODAN_API_KEY)
 
+
+import socket
+
 def shodan_org_scan(query: str) -> str:
-    if ("query" in query):
-        if "'" in query:
-            query = query.replace("'", '"')
-        x = query.split('"')
-        result = api.search(x)
+    report = ''
+    ipList = []
+
+    # Check if the query is a list of IPs
+    if ',' in query:
+        ips = query.split(',')
+        for ip in ips:
+            ip = ip.strip()  # remove any leading or trailing whitespace
+            host = api.host(ip, minify=True, history=False)
+            openPorts = []
+            for port in host['ports']:
+                openPorts.append(port)
```

Comment:
If `host['ports']` is a list, you can perform this variable assignment with `openPorts = host['ports']`. This way we can avoid using loop and append